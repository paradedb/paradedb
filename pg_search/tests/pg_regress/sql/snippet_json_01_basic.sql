\i common/snippet_json_basic_setup.sql

SELECT id,
       paradedb.snippet(metadata_jsonb->'details'->'author'->>'description'),
       paradedb.snippet_positions(metadata_jsonb->'details'->'author'->>'description')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');

SELECT id,
       paradedb.snippet(metadata_jsonb#>'{details,author,description}'),
       paradedb.snippet_positions(metadata_jsonb#>'{details,author,description}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');

SELECT id,
       paradedb.snippet(metadata_jsonb#>>'{details,author,description}'),
       paradedb.snippet_positions(metadata_jsonb#>>'{details,author,description}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');

SELECT id,
       paradedb.snippet(metadata_json->'tags'),
       paradedb.snippet_positions(metadata_json->'tags')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');

SELECT id,
       paradedb.snippet(metadata_json#>'{tags}'),
       paradedb.snippet_positions(metadata_json#>'{tags}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');

SELECT id,
       paradedb.snippet(metadata_json#>>'{tags}'),
       paradedb.snippet_positions(metadata_json#>>'{tags}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');

\i common/snippet_json_basic_cleanup.sql
