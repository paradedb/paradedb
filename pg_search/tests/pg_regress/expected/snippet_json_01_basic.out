\i common/snippet_json_basic_setup.sql
CREATE EXTENSION IF NOT EXISTS pg_search;
DROP TABLE IF EXISTS snippet_test;
CREATE TABLE snippet_test (
    id SERIAL PRIMARY KEY,
    content TEXT,
    metadata_json JSON,
    metadata_jsonb JSONB
);
INSERT INTO snippet_test (content, metadata_json, metadata_jsonb) VALUES
('This is a test test of the snippet function with multiple test words',
 '{"name": "test1", "tags": ["test", "snippet"], "metadata": {"created": "2023-01-01", "priority": 1}}',
 '{"id": 1, "details": {"author": {"first_name": "John", "last_name": "Doe", "description": "A test author"}, "stats": {"views": 100, "likes": 50}}, "active": true}'
),
('Another test of the snippet snippet function with repeated snippet words',
 '{"name": "test2", "scores": [10, 20, 30], "config": {"enabled": true, "settings": {"mode": "advanced", "limit": 5}}}',
 '{"id": 2, "nested": {"level1": {"level2": {"level3": "deep value"}}, "array": [1, "two", 3.14]}, "status": "active"}'
),
('Yet another test test test of the function function function',
 '{"name": "test3", "mixed": [{"key": "value"}, 42, null, true], "timestamp": "2023-12-31T23:59:59Z"}',
 '{"id": 3, "data": {"numbers": [1.1, 2.2, 3.3], "flags": {"debug": true, "test": false}}, "tags": ["alpha", "beta"]}'
),
('test Lorem ipsum dolor sit amet...test',
 '{"name": "test4", "complex": {"arrays": [[1,2], [3,4]], "object": {"null": null, "bool": false, "num": 3.14159}}}',
 '{"id": 4, "metadata": {"created_at": "2023-12-01", "updated_at": "2023-12-31", "versions": [1, 2, 3]}, "settings": {"notifications": {"email": true, "push": false}, "theme": "dark"}}'
);
CREATE INDEX ON snippet_test USING bm25 (
    id,
    metadata_json,
    metadata_jsonb
) WITH (
    key_field = 'id'
);
SELECT id,
       paradedb.snippet(metadata_jsonb->'details'->'author'->>'description'),
       paradedb.snippet_positions(metadata_jsonb->'details'->'author'->>'description')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');
 id |       snippet        | snippet_positions 
----+----------------------+-------------------
  1 | A <b>test</b> author | {"{2,6}"}
(1 row)

SELECT id,
       paradedb.snippet(metadata_jsonb#>'{details,author,description}'),
       paradedb.snippet_positions(metadata_jsonb#>'{details,author,description}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');
 id |       snippet        | snippet_positions 
----+----------------------+-------------------
  1 | A <b>test</b> author | {"{2,6}"}
(1 row)

SELECT id,
       paradedb.snippet(metadata_jsonb#>>'{details,author,description}'),
       paradedb.snippet_positions(metadata_jsonb#>>'{details,author,description}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_jsonb.details.author.description:test');
 id |       snippet        | snippet_positions 
----+----------------------+-------------------
  1 | A <b>test</b> author | {"{2,6}"}
(1 row)

SELECT id,
       paradedb.snippet(metadata_json->'tags'),
       paradedb.snippet_positions(metadata_json->'tags')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');
 id |       snippet       | snippet_positions 
----+---------------------+-------------------
  1 | test <b>snippet</b> | {"{5,12}"}
(1 row)

SELECT id,
       paradedb.snippet(metadata_json#>'{tags}'),
       paradedb.snippet_positions(metadata_json#>'{tags}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');
 id |       snippet       | snippet_positions 
----+---------------------+-------------------
  1 | test <b>snippet</b> | {"{5,12}"}
(1 row)

SELECT id,
       paradedb.snippet(metadata_json#>>'{tags}'),
       paradedb.snippet_positions(metadata_json#>>'{tags}')
FROM snippet_test
WHERE id @@@ paradedb.parse('metadata_json.tags:snippet');
 id |       snippet       | snippet_positions 
----+---------------------+-------------------
  1 | test <b>snippet</b> | {"{5,12}"}
(1 row)

\i common/snippet_json_basic_cleanup.sql
DROP TABLE IF EXISTS snippet_test;
